mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala

```diff
@@ -197,8 +238,86 @@ class LocalLDAModel private[clustering] (
   // TODO:
   // override def topicDistributions(documents: RDD[(Long, Vector)]): RDD[(Long, Vector)] = ???
 
+  /**
+   * Calculate and return per-word likelihood bound, using the `batch` of
+   * documents as evaluation corpus.
+   */
+  // TODO: calcualte logPerplexity over training set online during training, reusing gammad instead
+  // of performing variational inference again in [[bound()]]
+  def logPerplexity(
+      batch: RDD[(Long, Vector)],
+      totalDocs: Long): Double = {
```

Comment:
By average vs. total, I meant:
* average: perplexity normalized by number of docs in the sample.  E.g., if you double the number of docs by creating an extra copy of each one, then the perplexity should stay the same.
* total: perplexity without normalization (as it is now).  E.g., if you double the number of docs by creating an extra copy of each one, then the perplexity should double.

I prefer average since users can undo the normalization easily.